pytorch_lightning/plugins/sharded_plugin.py

```diff
@@ -43,7 +43,7 @@ def optimizer_state(self, optimizer: 'OSS') -> Optional[dict]:
         return self._optim_state_dict(optimizer)
 
     def on_before_forward(self, model: LightningModule, *args):
-        return model.transfer_batch_to_device(args, model.trainer.root_gpu)
+        return model._prepare_batch_for_transfer(args, model.trainer.root_gpu)
```

Comment:
just want to confirm this one. With sharded training one can override the `transfer_batch_to_device` hook but with DDPPlugin it's not possible right? @SeanNaren

Reply:
Yes! From what I understand DDP handles internally moving batches to devices. SDP allows the users to move to device manually hence why we override in the sharded plugin